src/mumble/ServerHandler.cpp

```diff
@@ -392,6 +393,11 @@ void ServerHandler::sendPing() {
 	if (!connection)
 		return;
 
+	if (iInFlightTCPPings >= g.s.iMaxInFlightTCPPings) {
+		serverConnectionClosed(QAbstractSocket::UnknownSocketError, tr("Server is not responding to TCP pings"));
```

Comment:
What I wanted to achieve here was to disconnect with a "reason".

Without a reason, the auto-reconnect code will not reconnect. (It treats empty reason and UnknownSocketError as "forced disconnect", I guess).

Also, the reason is nice to have in the disconnection message.

I initially had the same code here is used for timeout-during-connect (https://github.com/mkrautz/mumble/blob/e5da2ba6daa7c77823c2a47adb6476c584992d0f/src/mumble/ServerHandler.cpp#L525-L528). However, that turned out to not print any text, nor did it try to auto-reconnect.

What the current code does is pretty much equivalent to what ServerHandler::disconnect() does: it calls exit(0), which causes the connection to be torn down: https://github.com/mkrautz/mumble/blob/e5da2ba6daa7c77823c2a47adb6476c584992d0f/src/mumble/ServerHandler.cpp#L315-L340

Would you be happier if I introduced a `disconnectWithReason(QString reason)`, or added a `QString reason` parameter to ServerHandler's disconnect method?